src/components/NanoContract/NanoContractDetails.component.js

```diff
@@ -0,0 +1,211 @@
+    console.log('fetching more. After', after);
+    if (after == null) {
+      /* This situation is unlikely to happen because on the first transactions history load
+       * the `after` is assigned with the hash of the last transaction in the list.
+       */
+      return;
+    }
+    dispatch(nanoContractHistoryRequest({ ncId: nc.ncId, after }));
+  };
+
+  /* If an error happens on loading transactions history, an error feedback
+   * content must be presented to the user, and must have precedence over
+   * all other content.
+   */
+  const hasError = () => error != null;
+  const isEmpty = () => txHistory.length === 0 && !hasError();
+  const notEmpty = () => !isEmpty() && !hasError();
+
+  return (
+    <Wrapper>
+      <NanoContractDetailsHeader
+        nc={nc}
+        address={ncAddress}
+        onAddressChange={onAddressChange}
+      />
+      {isLoading && (
+        <Spinner style={styles.center} size={14} animating />
+      )}
+      {hasError()
+        && (<ErrorLoadingTransaction ncId={nc.ncId} error={error} />)}
+      {isEmpty()
+        && (<NoNanoContractTransaction />)}
+      {notEmpty()
+        && (
+          <HathorFlatList
+            data={txHistory}
+            renderItem={({ item }) => (
+              <NanoContractTransactionsListItem
+                item={item}
+                onPress={() => navigatesToNanoContractTransaction(item)}
+              />
+            )}
+            keyExtractor={(item) => item.txId}
+            refreshing={isLoading}
+            onRefresh={handleMoreTransactions}
+          />
+        )}
+    </Wrapper>
+  );
+};
+
+/**
+ * @param {Object} props
+ * @param {Object?} props.children Either a react component or a react element
+ */
+const Wrapper = ({ children }) => (
+  <View style={styles.wrapper}>
+    {children}
+  </View>
+);
+
+/**
+ * Renders a feedback for the lack of transactions.
+ */
+export const NoNanoContractTransaction = () => (
+  <FeedbackContent
+    title={t`No Nano Contract Transactions`}
+    message={t`There is not transaction to show yet.`}
+  />
+);
+
+/**
+ * Renders an error feedback for when the history load request goes wrong,
+ * and provides a call to action to try fetch the history again.
+ *
+ * @param {Object} props
+ * @param {string} props.ncId Nano Contract ID
+ * @param {string} props.error
+ */
+export const ErrorLoadingTransaction = ({ ncId, error }) => (
+  <FeedbackContent
+    title={t`Nano Contract Transactions Error`}
+    message={error}
+    icon={<Image source={errorIcon} style={{ height: 36, width: 36 }} resizeMode='contain' />}
+    action={<TryAgain ncId={ncId} />}
+  />
+);
+
+/**
+ * Renders a call to action to request history loading again.
+ *
+ * @param {Object} props
+ * @param {string} props.ncId Nano Contract ID
+ */
+export const TryAgain = ({ ncId }) => {
+  const dispatch = useDispatch();
+  const fetchTransactionsHistory = () => {
+    dispatch(nanoContractHistoryRequest({ ncId, after: null }));
+  };
```

Comment:
What happens if the second request fails? In this `Try Again` you are requesting from the beginning, so it would load the same transactions again.

Or you are deleting the history before this try again?

Reply:
This call gets from the beginning and I'm not cleaning the history. This is a bug. I'll fix it.

Reply:
Here: https://github.com/HathorNetwork/hathor-wallet-mobile/pull/438/files#r1626713412